### PR TITLE
Add support for podman

### PR DESCRIPTION
**Pull request by fgiloux**
Here are a  few modifications that allow to run the audit tool with podman instead of docker (docker stays the default). Therefore the user needs to set an environment variable: CONTAINER_ENGINE=podman before running the audit command.
I am not sure of the rational for this line in pkg/actions/get_bundle.go:
imageName := strings.Split(auditBundle.OperatorBundleImagePath, "@")[0]
It prevented podman commands to run successfully. Let me know if this has side effects I have overseen.

### CODE REVIEW

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
Hi @fgiloux, 

Your contributions are very welcome. 🥇  And that is on my to-do list for a while. 

We just need to do a few changes. Note that we are using cobra in the tool. In this way, the best approach is to create a new flag such as `--container-tool` to start to support it, see e.g: https://github.com/operator-framework/audit/blob/main/cmd/index/bundles/command.go#L82-L84

Also, we should add a new validation to ensure that the value informed is valid, see (e.g): https://github.com/operator-framework/audit/blob/main/cmd/index/bundles/command.go#L95

Then, we need to replace all places which have "docker" with the flag. Could you please address the changes? Also, please let me know if you need any help on that. We can do a par program and do that together.  

Note that we need to do that for all commands.  Btw, that could be another follow-up, however, would be very nice to do cleanup here and indeed centralize all common flags.

Again, really thank you for your contribution and I'd like you to know that see contributions here makes me very happy 🥇

**Comment by fgiloux** [comment]
Hi @camilamacedo86 
I have been thinking of using a flag instead of an environment variable and I think that the later provides a better user experience. Users will either use podman or docker. There is no reason to switch between one and the other. With an environment variable you set it once and you are done. With a flag you need to set it with every single command. You could create an alias but why would we want to request that from users when setting an environment variable could achieve the same?

**Comment by camilamacedo86** [comment]
Hi @fgiloux, 

> I have been thinking of using a flag instead of an environment variable and I think that the later provides a better user experience. 

Why the ENV VAR would be a better good experience? 
How as a user you could check you that this option exists? 
How hard would be if we have all values as ENV VAR? Is that a real environment variable or a usage option? 
How hard would be to ensure that we have doc all possible ENV VARs?
How hard would be keep maintained the logic based on ENV VARs for all commands?  

> There is no reason to switch between one and the other. With an environment variable, you set it once and you are done. With a flag, you need to set it with every single command.

If I want to use podman to login in to a register for report A and then use the docker to do report B? 
Also, we use COBRA and if you look at all tools that we have, for example, SDK and OPM this specific option is not informed by ENV VARs. We use the flags because of the reasons described above ( keep usability and maintainability ):

```
$ opm index add --help 
...
  -c, --container-tool string   tool to interact with container images (save, build, etc.). One of: [docker, podman]
```

```
 operator-sdk bundle validate --help 
...
 -b, --image-builder string                                 Tool to pull and unpack bundle images. Only used when validating a bundle image. One of: [docker, podman, none] (default "docker")
```

In this way, it seems more important we keep using the flags to:
- keep the maintainability: e.g. easier to ensure that will be documented via help 
- keep the usability: e.g. easier for users to know that this option exists since a tool user is expected to find all available options calling the --help and that. will not occur with the approach suggested, allow us to centralize the flags and ensure common behaviour across the project,  keep the expected standard across the board. 

> You could create an alias but why would we want to request that from users when setting an environment variable could achieve the same?

The ENV VAR does not achieve the same because: 
- does not provide a help option that describes how to use it and that is available (which is the expected behaviour for a command tool)
- does not allow me as a user to do different calls in the same ENV switching between them
- does to allow me as maintainer ensure that all is properly document
- does not allow me as a maintainer to use libs to generate automatic documentation for the based on the flags as we do in SDK
- does not allow me as a maintainer centralize its logic and ensure that all commands of the project have the same common options

Note for example, that if we get this PR merged it is not ensuring that all commands and places will use podman when you inform the value via ENV VAR. It has a lot of places that will still be calling docker.  

How could we achieve the use case `I'd like to define the option to use docker or podman and do not need to inform that in all commands`?

I'd be ok if we have the option via flag
Then, we doc in the flag help that this can be overwritten via ENV VAR
Then, in the cobra command (pre-run) method we would check if has the ENV VAR and set it in the flag value

In this way, we can allow the option via the flag, centralize the code that validate the flags and that overwrites the value. 

I hope that is helpful.

**Comment by fgiloux** [comment]
> How, as a user, you could check that this option exists without reading the docs?
In the command line usage there is a link to the doc :-). I have taken the approach you proposed and added both command lines and env vars. The command line description also mentions the env var option.

> How hard would it be if we had all values as ENV VAR?
I don't think it would be too hard, a bit time consuming. I am not sure it is useful for everything. The thing with the container engine is that you either use docker or podman. I don't think that people swap between one and the other for any other reason than making sure something runs with both and when they face an issue with one of them.

> How hard would it be to ensure that we have doc all possible ENV VARs?
Probably not that hard but my focus was supporting podman

> How hard would it be keep maintained the logic based on ENV VARs for all commands?
Probably not that hard but my focus was supporting podman
 
> If I want to use podman to log into a report A register and then use the docker to do report B?
As per my comment above I don't think it is a common scenario

> Also, we use COBRA, and if you look at all tools that we have, for example, SDK and OPM ENV VARs do not inform this specific option. We use the flags because of the reasons to keep usability and maintainability:
I tend to disagree on this one. Flags are perfectly fine for parameters specific to a command, which may change when the same command is called for something different. They are not suitable for environment configuration. That's why there is with kubectl both: flags, ~/.kube/config and KUBECONFIG env var. Some of the limitations of flags are highlighted [here](https://github.com/kubernetes-sigs/controller-runtime/blob/master/designs/component-config.md)
 
> > You could create an alias, but why would we want to request that from users when setting an environment variable that could achieve the same?
> 
> The ENV VAR does not achieve the same because:
> 
>     * does not provide a help option that describes how to use it and that is available (which is the expected behaviour for a command tool)
This is still possible to add the doc for environment variable  in the usage result. See for instance: $ kubectl config --help
 
>     * does not allow me as a user to do different calls in the same ENV switching between them
As per the above I don't think it is a valid use case for a container engine

>     * does not allow me as maintainer ensure that all is properly document
Why?
 
>     * does not allow me as a maintainer to use libs to generate automatic documentation for the based on the flags as we do in SDK
> 
>     * does not allow me as a maintainer to centralize its logic and ensure that all commands of the project have the same common options
Why? This is a design decision.

To summarize I have amended the PR to support both flags and environment variables. There are a couple of exceptions: the Makefile and the generators under hack where there is no COBRA.
I hope this better matches your expectations.

**Review by camilamacedo86 — APPROVED** [review]

**Review by camilamacedo86 — APPROVED** [review]
/lgtm
